applications/PfemFluidDynamicsApplication/custom_constitutive/fluid_laws/temperature_dependent/frictional_viscoplastic_temperature_dependent_3D_law.cpp

```diff
@@ -0,0 +1,141 @@
+
+//***********************************CLONE************************************
+//****************************************************************************
+
+ConstitutiveLaw::Pointer FrictionalViscoplasticTemperatureDependent3DLaw::Clone() const { return Kratos::make_shared<FrictionalViscoplasticTemperatureDependent3DLaw>(*this); }
+
+//*********************************DESTRUCTOR*********************************
+//****************************************************************************
+
+FrictionalViscoplasticTemperatureDependent3DLaw::~FrictionalViscoplasticTemperatureDependent3DLaw() {}
+
+std::string FrictionalViscoplasticTemperatureDependent3DLaw::Info() const { return "FrictionalViscoplasticTemperatureDependent3DLaw"; }
+
+//******************CHECK CONSISTENCY IN THE CONSTITUTIVE LAW******************
+//*****************************************************************************
+
+int FrictionalViscoplasticTemperatureDependent3DLaw::Check(const Properties& rMaterialProperties, const GeometryType& rElementGeometry,
+                        const ProcessInfo& rCurrentProcessInfo) const {
+
+    if (rMaterialProperties[DYNAMIC_VISCOSITY] < 0.0) {
+        KRATOS_ERROR << "Incorrect or missing DYNAMIC_VISCOSITY provided in process info for FrictionalViscoplasticTemperatureDependent3DLaw: "
+                     << rMaterialProperties[DYNAMIC_VISCOSITY] << std::endl;
+    }
+
+    if (rMaterialProperties[INTERNAL_FRICTION_ANGLE] < 0.0) {
+        KRATOS_ERROR << "Incorrect or missing INTERNAL_FRICTION_ANGLE provided in process info for FrictionalViscoplasticTemperatureDependent3DLaw: "
+                     << rMaterialProperties[INTERNAL_FRICTION_ANGLE] << std::endl;
+    }
+
+    if (rMaterialProperties[COHESION] < 0.0) {
+        KRATOS_ERROR << "Incorrect or missing COHESION provided in process info for FrictionalViscoplasticTemperatureDependent3DLaw: "
+                     << rMaterialProperties[COHESION] << std::endl;
+    }
+
+    if (rMaterialProperties[ADAPTIVE_EXPONENT] < 0.0) {
+        KRATOS_ERROR << "Incorrect or missing ADAPTIVE_EXPONENT provided in process info for FrictionalViscoplasticTemperatureDependent3DLaw: "
+                     << rMaterialProperties[ADAPTIVE_EXPONENT] << std::endl;
+    }
+
+    if (rMaterialProperties[BULK_MODULUS] <= 0.0) {
+        KRATOS_ERROR << "Incorrect or missing BULK_MODULUS provided in process info for FrictionalViscoplasticTemperatureDependent3DLaw: "
+                     << rMaterialProperties[BULK_MODULUS] << std::endl;
+    }
+
+    return 0;
+}
+
+double FrictionalViscoplasticTemperatureDependent3DLaw::GetEffectiveViscosity(ConstitutiveLaw::Parameters& rParameters) const {
+    return rParameters.GetConstitutiveMatrix()(5, 5);
+}
+
+double FrictionalViscoplasticTemperatureDependent3DLaw::GetEffectiveDensity(ConstitutiveLaw::Parameters& rParameters) const {
```

Comment:
same triplication of methods

Reply:
OK, thanks! I'll take a look next week, in the meanwhile I close the PR.